privacy/enforcement.go

```diff
@@ -23,6 +29,7 @@ func (e Enforcement) Apply(bidRequest *openrtb2.BidRequest) {
 
 func (e Enforcement) apply(bidRequest *openrtb2.BidRequest, scrubber Scrubber) {
 	if bidRequest != nil && e.Any() {
+		bidRequest = scrubber.ScrubRequest(bidRequest, e)
 		bidRequest.Device = scrubber.ScrubDevice(bidRequest.Device, e.getDeviceIDScrubStrategy(), e.getIPv4ScrubStrategy(), e.getIPv6ScrubStrategy(), e.getGeoScrubStrategy())
 		bidRequest.User = scrubber.ScrubUser(bidRequest.User, e.getUserScrubStrategy(), e.getGeoScrubStrategy())
 	}
```

Comment:
We can change this code to be a bit more precise using the `AnyLegacy` / `AnyActivities` and with overlap checks.

- We only need to run the new `ScrubRequest` if `AnyActivities` is true.
- I don't think we need to run `ScrubDevice` if `UFPD` and `PreciseGeo` already ran via `ScrubRequest`.
- I don't think we need to run `ScrubUser` if `UFPD`, `PreciseGeo` and `Eids` already ran via `ScrubRequest`.

Reply:
Done. Yes, good point